var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -353,7 +353,7 @@ class Gcc(AutotoolsPackage, GNUMirrorPackage):
     conflicts("%gcc@:4.7", when="@11:")
 
     # https://github.com/iains/gcc-12-branch/issues/6
-    conflicts("%apple-clang@14.0")
+    conflicts("%apple-clang@14.0.0")
```

Comment:
Is this not recovering the original meaning:
```suggestion
    conflicts("%apple-clang@14.0:")
```
?

Reply:
@alalazo I just tried this on my Intel Mac and it did not conflict, but tried to build GCC. 

That said it *worked* and actually built (using apple-clang 14.0.0). Should this instead be something like:
```python
    conflicts("%apple-clang@14.0.0", when="target=aarch64: platform=darwin")
```
? 

@adamjstewart was the issue only on M1/2?

Reply:
According to https://github.com/iains/gcc-12-branch/issues/6#issuecomment-1227519362 it's both Intel and Apple CPUs.

Reply:
> According to [iains/gcc-12-branch#6 (comment)](https://github.com/iains/gcc-12-branch/issues/6#issuecomment-1227519362) it's both Intel and Apple CPUs.

Huh. Well, this laptop is one where I have like...too many XCodes because I got hit with an XCode bug a while back. It's possible it's using some weird beta. Trust @iains waaaaay more than me with this. (It's a work laptop and things are working so I'm not uninstalling *anything* 😆 )

Reply:
Does this work?
```suggestion
    conflicts("%apple-clang@14:14.0")
```
I just bought a new M2 and it's supposed to be delivered today, so I should be able to test this in the future.

Reply:
Ahh!
```
❯ spack install gcc
==> Error: gcc conflicts with '%apple-clang@14:14.0'
```
That did seem to work!

Reply:
This is the syntax we used to have to use for all dependencies. Hopefully we can make compiler versions work like dependency versions in the future!